src/challenges.js

```diff
@@ -30,7 +57,23 @@ const cacheFunction = cb => {
 const reverseStr = str => {
   // reverse str takes in a string and returns that string in reversed order
   // The only difference between the way you've solved this before and now is that you need to do it recursivley!
+  let i = str.length--;
+
+  //let reversed = ''; // If it is here as it needs to be, are there one
+                       // too many function wrappings? Is it still closure?
+
+  const closureFunc = function(str) {
+    let reversed = '';
+    return (str, i) => {
+      if (i <= 0) return 0;
+      reversed += str[i];
+      return closureFunc(str, --i);
+    }
+  }
+  closureFunc(str, i);
+  return reversed;
 };
+console.log(reverseStr('qwerty'));
 
 const checkMatchingLeaves = obj => {
```

Comment:
Now that you've seen this implemented a time or two I think you can take a whack at it when you get some spare time.